### PR TITLE
Content add in

### PR DESCRIPTION
**Pull request by arttarawork**

### CODE REVIEW

**Comment by TCourtneyOwen** [comment]
Did we get confirmation that PPT and Excel are the only supported hosts?

**Comment by arttarawork** [comment]
> Did we get confirmation that PPT and Excel are the only supported hosts?

Yup! @keyur32 and @akrantz confirmed.

**Review by akrantz — CHANGES_REQUESTED** [review]
Please give me time to respond...

**Comment by akrantz** [comment]
OneNote Online also has support but not OneNote desktop.  Are we going to support that?

File: test/src/powerpoint-test-content.ts [hunk 1]
```diff
@@ -14,25 +14,27 @@ Office.onReady(async (info) => {
 });
 
 export async function runTest(): Promise<void> {
-    return new Promise<void>(async (resolve, reject) => {
+    return new Promise<void>((resolve, reject) => {
```
Comment (akrantz) [inline test/src/powerpoint-test-content.ts:17]: Please change back to use await. Thanks.

File: manifest.xml [hunk 2]
```diff
@@ -22,184 +21,7 @@
     <Host Name="Workbook"/>
   </Hosts>
   <DefaultSettings>
-    <SourceLocation DefaultValue="https://localhost:3000/taskpane.html"/>
+    <SourceLocation DefaultValue="https://localhost:3000/content.html"/>
   </DefaultSettings>
   <Permissions>ReadWriteDocument</Permissions>
-  <VersionOverrides xmlns="http://schemas.microsoft.com/office/taskpaneappversionoverrides" xsi:type="VersionOverridesV1_0">
```
Comment (TCourtneyOwen) [inline manifest.xml:28]: We need to add other supported hosts here.  Definitely PPT and maybe Word if it's supported

File: test/test-manifest.xml [hunk 3]
```diff
@@ -22,49 +22,7 @@
     <Host Name="Workbook"/>
```
Comment (TCourtneyOwen) [inline test/test-manifest.xml:22]: We need to add PPT and other supported hosts (if any) to the manifest

File: convertToSingleHost.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 const convertTest = process.argv[3] === "convert-test";
 const fs = require("fs");
```
Comment (TCourtneyOwen) [inline convertToSingleHost.js:2]: Please incorporate latest changes to convertToSingleHost that were just added at https://github.com/OfficeDev/Office-Addin-TaskPane/commit/57709ce2558d4777243f7d22d60b7ac420ec49d8#diff-d66dc199c22efc5aae8a395f81d475b407fbfcb23f843903241caaf6f373ebe7

File: test/src/powerpoint-test-content.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    if (info.host === Office.HostType.PowerPoint) {
+        const testServerResponse: object = await pingTestServer(port);
+        if (testServerResponse["status"] == 200) {
+            await runTest();
+        }
+    }
+});
+
+export async function runTest(): Promise<void> {
+    return new Promise<void>(async (resolve, reject) => {
```
Comment (akrantz) [inline test/src/powerpoint-test-content.ts:17]: Also this code shouldn't use `new Promise()` with resolve() and reject(). An async function resolves when it reaches the end and rejects by throwing an exception.

Comment (akrantz) [reply test/src/powerpoint-test-content.ts:17]: You can remove the try/catch and just allow the exception to propogate.

File: test/src/powerpoint-test-content.ts [hunk 6]
```diff
@@ -14,27 +14,25 @@ Office.onReady(async (info) => {
...
-                                testHelpers.addTestResult(testValues, "output-message", asyncResult.value, " Hello World!");
-                            }
-                            await sendTestResults(testValues, port);
-                            testValues.pop();
-                            resolve();
-                        });
-                    });
-                });
+            await run();
+            await run();
```
[the remark this answers is not in the document]
Comment (arttarawork) [reply test/src/powerpoint-test-content.ts:21]: Weird behavior with getSelectedDataAsync(). Running run() once adds "Hello World!" to the slide, but doesn't select the new text properly, so running getSelectedDataAsync() after running run() once returns "". Running run() twice inserts "Hello World! Hello World!". At this point, the latter part (" Hello World!") is what's selected, which is what getSelectedDataAsync() reads. 

It's a workaround I don't like, so if there's a proper way to select that text, I would use it, but I haven't found anything that lets me do that.

File: test/src/powerpoint-test-content.ts (hunk 5 above)
Comment (arttarawork) [reply test/src/powerpoint-test-content.ts]: Let me know if the recent changes are satisfactory for that

File: src/content/powerpoint.ts [hunk 7]
```diff
@@ -22,15 +22,8 @@ export async function run() {
...
-        console.error(result.error.message);
-      }
-    }
-  );
-}
+  const options: Office.SetSelectedDataOptions = { coercionType: Office.CoercionType.Text };
+
+  await Office.context.document.setSelectedDataAsync(" ", options);
+  await Office.context.document.setSelectedDataAsync("Hello World!", options);
+}
```
Comment (akrantz) [inline src/content/powerpoint.ts:29]: need a newline at the end of the last line...

**Comment by akrantz** [comment]
See https://github.com/OfficeDev/Office-Addin-Content/pull/1

**Comment by TCourtneyOwen** [comment]
@Artur, do you want to try moving this branch over to https://github.com/OfficeDev/Office-Addin-Content so we can then see if it passes the CI job?

**Comment by arttarawork** [comment]
> @artur, do you want to try moving this branch over to https://github.com/OfficeDev/Office-Addin-Content so we can then see if it passes the CI job?

The contents of this branch should be already included in this PR: https://github.com/OfficeDev/Office-Addin-Content/pull/1

@akrantz are we good to merge that PR?

**Comment by TCourtneyOwen** [comment]
One thing I would check out is that linting passes. I noticed in a trial run of the CI pipeline against the commit that Adam made to populate the new repo that the linting phase failed in the CI job

**Comment by akrantz** [comment]
This branch is already part of https://github.com/OfficeDev/Office-Addin-Content/pull/1. Let's comment about any remaining work like fixing linting issues there.

**Comment by arttarawork** [comment]
> One thing I would check out is that linting passes. I noticed in a trial run of the CI pipeline against the commit that Adam made to populate the new repo that the linting phase failed in the CI job

True, but the first PR in that branch is supposed to be the actual content add-in. Is there a way to set up pipelines for PRs? I think I can manually run the pipeline for now from the ADO menu

**Comment by arttarawork** [comment]
^ https://office.visualstudio.com/OC/_build/results?buildId=7252873&view=results

**Comment by TCourtneyOwen** [comment]
The pipeline is hooked up already.  As long as your PR is against main it will run

**Pull request closed without merging**